src/shims/unix/foreign_items.rs

```diff
@@ -236,6 +237,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             "open" | "open64" => {
                 // `open` is variadic, the third argument is only present when the second argument has O_CREAT (or on linux O_TMPFILE, but miri doesn't support that) set
                 this.check_abi_and_shim_symbol_clash(abi, Conv::C , link_name)?;
+                this.check_fixed_args_count("open/open64", abi, 2)?;
```

Comment:
Now it's still fairly easy to forget to call `check_fixed_args_count`.

So instead, please add a new `this.check_shim_variadic` that is a lot like `check_shim` but returns something like `(&'a [OpTy<'tcx>; N], &'a [OpTy<'tcx>])` where the "fixed" arguments are mapped to the array and the variadic ones are returned in the slice.